### ISSUE TITLE
transmission-remote JSON output problem

### ISSUE THREAD

**Issue by SNBogdanov** [transmission/transmission#7045]
### What is the issue?

When I asked for json output for such request
transmission-remote  -t 1  -j -i -f
I received 2 rows 
One for {"arguments":{"torrents":[{"activityDate"
another is fo {"arguments":{"torrents":[{"files"
It will be nice to have one response with all data in one place
With file info you don't have torrent name and id, so you need to "merge" two arrays to have complete data.

### Which application of Transmission?

transmission-remote

### Which version of Transmission?

4.1.0-dev

**Comment by killemov** [comment]
AFAICT this is just weird, my guess is that transmission-remote does 2 separate requests.

**Comment by valcomm** [comment]
transmission-remote uses RPC.  There is no commands combining in RPC (nor should be) - hence two separate requests.

**Comment by killemov** [comment]
Which requests are those again?

**Comment by valcomm** [comment]
> Which requests are those again?

There's a --debug option to get your answer

**Comment by SNBogdanov** [comment]
This information can be received in one request, just put 2 field in arguments

**Comment by valcomm** [comment]
Yes, it could be.  But you're using a client with its own functionality.  And those two pieces of information are printed out separately.  No real need to make such a simple client too smart (with a lot of cases like one torrent or many for example).  
If you want to combine them - nothing stops you from calling RPC directly.

**Comment by SNBogdanov** [comment]
Simple action. Combine all options in one request. It will work for multiple torrents too

**Comment by valcomm** [comment]
You're missing the point.  transmission-remote isn't a RPC calls executor.  It is a simple client, that allows to get its output either printed or as JSON.  That's it.

**Comment by killemov** [comment]
If it is about 2 separate `torrent-get` calls for the same torrent then they clearly should be combined.

**Comment by valcomm** [comment]
Why?  And especially "clearly"?  What's the gain with combining them by extremely complicating options parsing, executing RPC and then parsing JSON to output that in the options order (e.g. not as JSON, options might be something like -t 1 -j -i -ph 1 -G 0 -f).
I think it's better if it's kept the simple and predictable way it is now: one option - one RPC execution.  No need to complicate the simple client.  
If someone wants something tricky that covers some his/her specific cases - there are RPC libraries for many languages.

**Comment by SNBogdanov** [comment]
Simple and predictable?
try ask for -t 1,2 -f -i
you receive 2 arrays of torrents and in array for files where is no information about torrent id to match files structure to torrent itself

**Comment by valcomm** [comment]
Arrays are ordered, right?  Is it so hard to match 0th element of the first array with 0th element of the second array?  What other information do you need for that beside element indexes?

**Comment by killemov** [comment]
Also, using table format for a single torrent may not be desirable, so forcing output to "classic" is possible, correct?

**Comment by valcomm** [comment]
> Also, using table format for a single torrent may not be desirable, so forcing output to "classic" is possible, correct?

If that was a question for me, I don't understand what are you asking about.  "table format", "desirable", "classic" - am I supposed to guess what do you mean/prefer here?

Again: transmission-remote is a simple client that lets you to get received data as JSON.
If someone needs something more complicated/powerful - there's a pretty well described RPC protocol.

**Comment by SNBogdanov** [comment]
Having two separate arguments row have another drawback.
It cannot be readed by json converters since it is not combined in array of arguments

**Comment by killemov** [comment]
@valcomm "classic" as in before table format, meaning a single object containing all fields instead of two matching arrays. You obviously have an agenda keeping things the way they are now.

@SNBogdanov You are correct. It can't be done unless your converter can merge the resulting objects. When you request multiple fields from a single torrent through a single command line execution, I would call returning that as more than one result a bug.

**Comment by valcomm** [comment]
> Having two separate arguments row have another drawback.

No, it does not.  transmission-remote is a simple client that let's you see the output as JSON (recently added feature).
If you need a flexible RPC executor that provides you JSON - either write one yourself or find someone to do that (that's not that hard and even examples in bash were provided in some other discussion here).

@killemov, should that table format (I still have no idea what's that) be used for calls like -t 1 -j -i -ph 1 -G 0 -f ?  Could you describe how this case is different in simple words for the documentation/man page?

**Comment by killemov** [comment]
@valcomm See https://github.com/transmission/transmission/blob/main/docs/rpc-spec.md#33-torrent-accessor-torrent-get and https://github.com/killemov/Shift/blob/3be3acb2f3b15359267ba93015bee1b6c95d90ea/shift.js#L3315

The table format is a bandwidth saving format that should not be used for a single torrent.

**Comment by valcomm** [comment]
> The table format is a bandwidth saving format that should not be used for a single torrent.

Could you elaborate on that?  What damage [in functionality] hits transmission-remote when bandwidth saving format is used for a single torrent by this client?  I'm all ears.

**Comment by killemov** [comment]
> Could you elaborate on that?

No.
> What damage [in functionality] hits transmission-remote when bandwidth saving format is used for a single torrent by this client?

Because the receiving client (/user) has to match an array of field names with an array of field data instead of having a single JSON object ready to go. And for a single torrent it's actually more overhead.
> I'm all ears.

...

**Comment by tearfur** [comment]
> parsing JSON to output that in the options order

I think this is the biggest difficulty in implementing this feature. But the idea itself is good, so if anyone can write code that:

1. Isn't (much) more complicated than the current code.
2. Prints the output in the order of the command line options.

Then I'd absolutely welcome it.
